chore: Refactor icon components (#2615)
Workday/canvas-kit: modules/docs/mdx/11.0-UPGRADE-GUIDE.mdx

🤖 The codemod will handle the change of `hasError` to `error` for you automatically.

### Icons

> All icon components, `Svg`, `SystemIcon`, `SystemIconCircle`, `AccentIcon`, `AppletIcon`, `Graphic`,

Reviewer: ```suggestion
All icons components including `Svg`, `SystemIcon`, `SystemIconCircle`, `AccentIcon`, `AppletIcon` and `Graphic`
```

> now use

Reviewer: ```suggestion
now use our new
```

[Canvas Tokens](https://workday.github.io/canvas-tokens/?path=/docs/docs-getting-started--docs) and
our
[new styling utilities](https://workday.github.io/canvas-kit/?path=/docs/styling-basics--create-modifiers#createstyles-api).

> The component now supports the `cs` prop, but otherwise the API has not changed. It should behave

Reviewer: ```suggestion
These components also supports the new `cs` prop, but otherwise the API has not changed. It should behave
```

> identically as it did in previous versions.

Reviewer: i guess since we removed the styles prop, this isn't exactly true, is this a breaking change?

> The `styles` prop has been removed from icon components.

Reviewer: ```suggestion
**Note:The `styles` prop has been removed from icon components.**
```

🤖 The codemod will handle the change of `styles` to `cs` for you automatically.

#### AccentIcon

`AccentIconStyles` prop has been deprecated and will be merged with `AccentIconProps` in a future
version. `accentIconStyles` has been deprecated and will be removed in a future version as a part of
implementation of stencils and new tokens. Consider to use `accentIconStencil` to get styles or
class name.

#### AppletIcon

> `AppletIconStyles` prop has been deprecated and will be merged with `AppletIconProps` in a future

Reviewer: I think you can break these out into bullet points:
```
- `AppletIconStyles` prop has been deprecated and will be merged with `AppletIconProps` in a future major version.
- `appletIconStyles` has been deprecated and will be removed in a future version as a part of
implementation of stencils and new tokens.
```

version. `appletIconStyles` has been deprecated and will be removed in a future version as a part of

> implementation of stencils and new tokens. Consider to use `appletIconStencil` to get styles or

Reviewer: ```suggestion
implementation of stencils and new tokens. Consider using `appletIconStencil` to get styles or
```

class name.

#### Graphic

`GraphicStyles` prop has been deprecated and will be merged with `GraphicProps` in a future version.
`graphicStyles` has been deprecated and will be removed in a future version as a part of

> implementation of stencils and new tokens. Consider to use `graphicStencil` to get styles or class

Reviewer: I think we should give an example of how to get the className or styles, also not sure if this should be part of the PR? Are we ready to show the use of stencils and are we exporting them? cc @alanbsmith @NicholasBoll

name.

#### Icon

> Internal only. `Icon` component has been removed and no longer used in icons. It was replaced by

Reviewer: What does internal only mean?

Author: Canvas Kit only, this component has n't been exported
